File: js_checkio_solutions/Scientific Expedition/caps_lock.js
#!/usr/bin/env checkio --domain=js run caps-lock
"use strict";
// Joe Palooka has fat fingers, so he always hits the “Caps Lock” key whenever he actually intends to hit the key  “a” by itself. (When Joe tries to type in some accented version of “a” that needs more keystrokes to conjure the accents, he is more careful in the presence of such raffinated characters ([Shift] + [Char]) and will press the keys correctly.) Compute and return the result of having Joe type in the given text. The “Caps Lock” key affects only the letter keys from “a” to “z” , and has no effect on the other keys or characters. “Caps Lock” key is assumed to be initially off.
// 
// Input:A string. The typed text.
// 
// Output:A string. The showed text after being typed.
// 
// The mission was taken from Python CCPS 109 Fall 2018. It is taught for Ryerson Chang School of Continuing Education byIlkka Kokkarinen
// 
// 
// END_DESC
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var assert_1 = __importDefault(require("assert"));
var log = console.log;
function capsLock(text) {
    var isCaplocked = false;
    //will use text as input string and push changes to output below
    var output = [];
    for (var i = 0; i < text.length; i++) {
        if (text[i] === 'a') {
            isCaplocked = !isCaplocked;
        }
        else {
            (isCaplocked) ? output.push(text[i].toUpperCase()) : output.push(text[i]);
        }
    }
    return output.join('');
}
console.log('Example:');
console.log(capsLock('Why are you asking me that?'));
if (true) {
    // These "asserts" are used for self-checking
    assert_1.default.equal(capsLock('Why are you asking me that?'), 'Why RE YOU sking me thT?');
    assert_1.default.equal(capsLock('Always wanted to visit Zambia.'), 'AlwYS Wnted to visit ZMBI.');
    console.log("Coding complete? Click 'Check' to earn cool rewards!");
}

File: py_checkio_solutions/Scientific Expedition/caps_lock.py
#!/usr/bin/env checkio --domain=py run caps-lock

# Joe Palooka has fat fingers, so he always hits the “Caps Lock” key whenever he actually intends to hit the key  “a” by itself. (When Joe tries to type in some accented version of “a” that needs more keystrokes to conjure the accents, he is more careful in the presence of such raffinated characters ([Shift] + [Char]) and will press the keys correctly.) Compute and return the result of having Joe type in the given text. The “Caps Lock” key affects only the letter keys from “a” to “z” , and has no effect on the other keys or characters. “Caps Lock” key is assumed to be initially off.
# 
# Input:A string. The typed text.
# 
# Output:A string. The showed text after being typed.
# 
# The mission was taken from Python CCPS 109 Fall 2018. It is taught for Ryerson Chang School of Continuing Education byIlkka Kokkarinen
# 
# 
# END_DESC

def caps_lock(text: str) -> str:
    
    CAP_LOCK = False

    ans = ''

    for n in text:
        if n == 'a':
            if CAP_LOCK:
                CAP_LOCK = False
            else:
                CAP_LOCK = True
        else:
            if CAP_LOCK:
                ans += n.upper()
            else:
                ans += n

    return ans


if __name__ == '__main__':
    print("Example:")
    print(caps_lock("Why are you asking me that?"))

    # These "asserts" are used for self-checking and not for an auto-testing
    assert caps_lock("Why are you asking me that?") == "Why RE YOU sking me thT?"
    assert caps_lock("Always wanted to visit Zambia.") == "AlwYS Wnted to visit ZMBI."
    print("Coding complete? Click 'Check' to earn cool rewards!")